From a21175a1631febadfddea1c7639c8cf998559f13 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Thu, 17 Mar 2016 14:58:14 +0800
Subject: [PATCH] ARM64: dts: rockchip: add wdt0 for rk3399

There are two watchdogs in ALIVE named WDT0 and WDT1, and
one watchdog in PMU named WDT2.

WDT0 can drive CRU to generate global software reset.

Change-Id: Ide47e7e69572d2f2a537b590dc75010cf0f56c51
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3399.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index fe4429a7eba3..8a5194bac8ec 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -758,6 +758,14 @@
 		reg = <0x0 0xff770000 0x0 0x10000>;
 	};
 
+	wdt0: watchdog@ff840000 {
+		compatible = "snps,dw-wdt";
+		reg = <0x0 0xff840000 0x0 0x100>;
+		clocks = <&cru PCLK_WDT>;
+		interrupts = <GIC_SPI 120 IRQ_TYPE_LEVEL_HIGH>;
+		status = "disabled";
+	};
+
 	spdif: spdif@ff870000 {
 		compatible = "rockchip,rk3399-spdif";
 		reg = <0x0 0xff870000 0x0 0x1000>;
-- 
2.35.3

